### DISCUSSION TITLE
Syntax for selecting nth element of an array

### DISCUSSION THREAD

**Discussion by DonBranson** [Kong/insomnia#3300] [Ideas]
In a request using chaining, I want to extract the zeroth element of an array. This syntax returns an array:

```
$.content[?(@.content.email == 'dr.smith@example.com')]
```

Saying,

```
Returned more than one result: $.content[?(@.content.email == 'dr.smith@example.com')]
```

I've tried a few things to no avail, but expected this syntax would select the zeroth element:

```
Returned more than one result: $.content[?(@.content.email == 'dr.smith@example.com')][0]
```

But the Live Preview says,

```
Returned no results: $.content[?(@.content.email == 'dr.smith@example.com')][0]
```

What's the correct syntax?

**Comment by develohpanda** [comment]
Hmm, I'm not entirely sure about what the correct syntax should be. Does your query work in https://jsonpath.com/ if you try with some sample data?

There are some [related tickets](https://github.com/Kong/insomnia/issues/777) to explore moving to `jq` and this may be an issue with `jsonpath` itself. cc: @dimitropoulos

**Reply by DonBranson** [reply]
Thanks for the recommendation of jsonpath - it's also showing an issue with my syntax. Using this sample data:

```
{
  "content": [
    {
      "content": {
        "firstName": "Dr.",
        "lastName": "Smith",
        "phoneNumber": "yep",
        "email": "dr.smith@example.com"
      }
    },
    {
      "content": {
        "firstName": "Dr.",
        "lastName": "Smith",
        "phoneNumber": "yep",
        "email": "dr.smith@example.com"
      }
    },
    {
      "content": {
        "firstName": "Bob",
        "lastName": "Smith",
        "phoneNumber": "yep",
        "email": "dr.smith@example.com"
      }
    }
  ]
}
```

The query `$.content[?(@.content.email == 'dr.smith@example.com')]` returns an array of three, so it seems like `[0]` is the code to select the zeroth item, but I am perhaps putting it in the wrong place or missing something.

**Reply by DonBranson** [reply]
I also tried functions like .first(), .min(), etc.

**Reply by develohpanda** [reply]
Interesting, I did some testing here as well and it seems to be a limitation of JSONPath itself 😞 Maybe the request chaining editor needs to be expanded with an option to programmatically select the nth result?

**Reply by DonBranson** [reply]
I'm open to that. I started to look at the code to see what json path parser you're using, then got distracted by other tasks. :)  Is it https://github.com/dchester/jsonpath? It appears to support subscripting, but somehow the array returned by a filter isn't the right kind of array. Hmmm.

**Reply by develohpanda** [reply]
Yeah that's the one: https://www.npmjs.com/package/jsonpath. There is talk about moving away from jsonpath but there isn't a concrete plan or timeline yet.

**Reply by DonBranson** [reply]
Gotcha. I asked this question over there: [Syntax for selecting nth element of an array from a filter response](https://github.com/dchester/jsonpath/issues/157).

**Reply by develohpanda** [reply]
Sounds good, let's see what they come back with. If nothing else I don't think adding a nth element option will be too difficult

**Reply by DonBranson** [reply]
They haven't responded to my question just yet. I'd like an nth element option when you have a chance.

I have an unrelated question that could probably start a new thread, but...is there a way to create a script-like thing that just says, "hey, run these 20 requests"? I can start a new thread for this if you prefer.

**Reply by develohpanda** [reply]
Sorry about the delay! I haven't been able to get to exposing an option like that yet. For the next question, could you open a new discussion for that please? 🙏🏽

**Reply by DonBranson** [reply]
Yes, and sorry about my delay. :)
